server/apps/archive_broadcast/broadcast.py

```diff
@@ -0,0 +1,142 @@
+        genre_list = get_resource_service('vocabularies').find_one(req=None, _id='genre') or {}
+        broadcast_genre = [genre for genre in genre_list.get('items', []) if genre.get('value') == BROADCAST_GENRE]
+
+        if not broadcast_genre:
+            raise SuperdeskApiError.badRequestError(message="Cannot find the {} genre.".format(BROADCAST_GENRE))
+
+        doc['broadcast'] = {'status': ''}
+        doc['genre'] = broadcast_genre
+        doc['family_id'] = item_id
+
+        for key in FIELDS_TO_COPY:
+            doc[key] = item.get(key)
+
+        resolve_document_version(document=doc, resource=SOURCE, method='POST')
+        service.post(docs)
+        insert_into_versions(id_=doc[config.ID_FIELD])
+        build_custom_hateoas(CUSTOM_HATEOAS, doc)
+        return [doc[config.ID_FIELD]]
+
+    def _valid_broadcast_item(self, item):
+        """
+        Broadcast item can only be created for Text or Preformatted item.
+        The state of the item cannot be Killed, Scheduled or Spiked
+        :param dict item: item from which the broadcast item will be created
+        """
+        if not item:
+            raise SuperdeskApiError.notFoundError(
+                message="Cannot find the requested item id.")
+
+        if not item.get(ITEM_TYPE) in [CONTENT_TYPE.TEXT, CONTENT_TYPE.PREFORMATTED]:
+            raise SuperdeskApiError.badRequestError(message="Invalid content type.")
+
+        if item.get(ITEM_STATE) in [CONTENT_STATE.KILLED, CONTENT_STATE.SCHEDULED, CONTENT_STATE.SPIKED]:
+            raise SuperdeskApiError.badRequestError(message="Invalid content state.")
+
+    def is_broadcast(self, item):
+        """
+        Item to check if broadcast or not.
+        :param dict item:
+        :return: If broadcast then true else false
+        """
+        return item.get('genre') and any(genre.get('value') == BROADCAST_GENRE for genre in item.get('genre', []))
+
+    def enhance_items(self, items):
+        """
+        Sets the broadcast_id attribute if master story has broadcast script
+        :param list items: list of items
+        :return:
+        """
+        ids = [item.get(config.ID_FIELD) for item in items
+               if item.get(ITEM_TYPE) in [CONTENT_TYPE.TEXT, CONTENT_TYPE.PREFORMATTED]]
+
+        query = {
+            'query': {
+                'filtered': {
+                    'filter': {
+                        'and': [
+                            {'terms': {'family_id': ids}},
+                            {'term': {'genre.name': BROADCAST_GENRE}}
+                        ]
+                    }
+                }
+            }
+        }
+
+        request = ParsedRequest()
+        request.args = {'source': json.dumps(query)}
+        broadcast_items = get_resource_service(SOURCE).get(req=request, lookup=None)
+
+        for broadcast_item in broadcast_items:
+            item = next((item for item in items if item.get(config.ID_FIELD) == broadcast_item.get('family_id')))
```

Comment:
Just checking - is it possible that the generator expression does not yield any results here? This would result in an error, because there is no default value argument passed to the `next()` function.